lib/computer_player.rb

```diff
@@ -46,27 +47,38 @@ def minimax(depth, board, current_mark, alpha, beta)
   end
 
   def score_for_move(board, depth)
-    if !board.has_winner?
-      return 0
-    elsif board.winner == computer_mark
-      return depth
+    if !board.winner?
+      0
+    elsif board.winner_mark == computer_mark
+      depth
     else
-      return - depth
+      -depth
     end
   end
 
   def switch_mark(mark)
-    mark = mark == :X ? :O : :X
+    mark = mark == X ? O : X
+  end
+
+  def ready?
+    true
   end
 
   private
 
   def score_favourable_for_computer?(current_mark, score, best_score)
-    current_mark == computer_mark && score[0] > best_score ||
-      current_mark != computer_mark && score[0] < best_score
+    if (computer_mark_is?(current_mark))
```

Comment:
The question mark (pun!) breaks the readability... How about simply computer?(current_mark)